server/src/api/community.rs

```diff
@@ -417,42 +420,89 @@ impl Perform for Oper<EditCommunity> {
       Err(_e) => return Err(APIError::err("couldnt_update_community").into()),
     };
 
-    // Mod tables
-    if let Some(removed) = data.removed.to_owned() {
-      let expires = match data.expires {
-        Some(time) => Some(naive_from_unix(time)),
-        None => None,
-      };
-      let form = ModRemoveCommunityForm {
-        mod_user_id: user_id,
+    // TODO there needs to be some kind of an apub update
+    // process for communities and users
+
+    let edit_id = data.edit_id;
+    let community_view = blocking(pool, move |conn| {
+      CommunityView::read(conn, edit_id, Some(user_id))
+    })
+    .await??;
+
+    let res = CommunityResponse {
+      community: community_view,
+    };
+
+    if let Some(ws) = websocket_info {
```

Comment:
You have this identical code copy pasted into three places in this file. And a bunch more places that do the `if let Some(ws)` and `chatserver.do_send()`. Please extract that into a single function.

Reply:
If you look at them, they're all sending different messages (`DeleteCommunity`, `EditCommunity`, `RemoveCommunity`), and there are a few other `if let Some(ws)` in that file that have a lot of different forms too. 

But also, this method is already genericized as much as it can be:

```
      ws.chatserver.do_send(SendCommunityRoomMessage { // The typed context
        op: UserOperation::BanFromCommunity, // the op
        response: res.clone(), // Could be any json
        community_id: data.community_id,
        my_id: ws.id,
      });
```

Reply:
I mean there are 3 different functions that return `CommunityResponse`, so for those you could easily extract everything from `CommunityView::read()` down to `chat_server.send()`. Just need to pass in the op and community_id.

Reply:
Okay just did this.